pkg/content/file.go

```diff
@@ -138,10 +221,26 @@ func (s *FileStore) Writer(ctx context.Context, opts ...content.WriterOpt) (cont
 		}
 	}
 
-	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
-		return nil, err
+	var (
```

Comment:
The format feels a bit like two funcs. Do think it will look better if moving the validation before this line into a func something like validateWriteOptions?

Reply:
refactored to three funcs